### DISCUSSION TITLE
Prevent asking for timezone

### DISCUSSION THREAD

**Discussion by jasonvarga** [studio1902/statamic-peak#406] [Ideas]
Edit: Ignore this request, at least for now. Once Statamic natively handles timezones better, *then* it would make sense to avoid changing your timezone.

---

Hi!

During the setup, there is a prompt that asks for your site's timezone. I recommend that you remove that and just use UTC.

Or, if you must show the prompt, at least default to UTC so you can just hit enter.

It's best practice to use UTC and I feel the prompt is encouraging people to go against it even if that wasn't the intention.

At Statamic we're going to be fighting an uphill battle to convert to UTC because of going against the best practice years ago. 
https://github.com/statamic/ideas/issues/842

**Comment by robdekort** [comment]
I know it's against best practices, but there's currently no way in Statamic to get stuff like post dates and scheduled times to properly reflect a users timezone, as far as I know that is. This is why it's in there, and I why I don't currently use UTC for my sites. 

Or should this be handled differently?

**Reply by marcorieser** [reply]
Same here

**Reply by jasonvarga** [reply]
Scheduling works as long as you enter the dates in UTC, which is obviously counter intuitive when you click "now" in the CP and expect it to mean now in your time.

It's a catch 22 because if we want to improve support for user timezones, we'll need to save them in UTC. It's going to a pain to migrate if you're storing them in a local timezone.

You can feel free to ignore this.

**Reply by robdekort** [reply]
It would make zero sense for my clients (CP users) to have to use UTC. They probably don't even know what it is. They're going to want to use their own local time and expect that to work. As long as it doesn't, I'm afraid I'm going to have to stick to going against best practices. Sorry!

It has always been my intention of removing this from the installer when the CP supports time conversions to a users timezone.

**Reply by robdekort** [reply]
I could default to UTC better in the prompt btw. That's a great call, if prompts supports that. Probably!

**Reply by marcorieser** [reply]
@jasonvarga I probably don't see all the critical bits but my simple approach for migration would be:
1. check if the timezone is something other than UTC
2. if so, find all the date fields and configure them correctly (programmatically of course)
3. batch save all the entries
4. done

**Reply by robdekort** [reply]
Darn, I think my memory was right for once, you can't set a default in a Laravel search prompt.

**Reply by jasonvarga** [reply]
We don't need to clutter the peak repo with this, but yes, thanks that's about the gist of it. Of course it's simpler to write a bullet list than to implement though. 😄

**Reply by robdekort** [reply]
I think Marco is offering to write it for you...

**Comment by jasonvarga** [comment]
You know what, ignore this. It's our burden to deal with. No sense making it more difficult for users in the mean time. 👌
